10-28-20/fib.swift

```diff
@@ -0,0 +1,18 @@
+func fib(n: Int) -> Int {
+  var newMemo:[Int:Int] = [:]
+  return _fib(n: n, memo: &newMemo)
+}
+
+func _fib(n: Int, memo: inout Dictionary<Int, Int>) -> Int {
+  guard memo[n] == nil else { return memo[n] ?? -1 }
+  guard n > 2 else { return 1 }
+
+  memo[n] = _fib(n: n - 1, memo: &memo) + _fib(n: n - 2, memo: &memo)
+  return memo[n] ?? -1
```

Comment:
So swift is complaining here because accessing from the memo is going to either return nil or an `Int`. One way to get around this is to provide a default value using `??` as you've done. Another is to use `!` which tells swift that you're *certain* that this returns a proper value.

So my suggestion is to either write `memo[n]!` here, or just save `_fib(n: n - 1, memo: &memo) + _fib(n: n - 2, memo: &memo)` into another variable and you can return that variable at the end.

Reply:
Ahh right on. Figured there would be some way to not use some arb value. I'll go with force unwrap. Thanks @EricChiu920